src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -265,15 +288,24 @@ private CallableStatement getCallableStatementHandle(CallableHandles request,
             String catalog) throws SQLServerException {
         CallableStatement CS = null;
         HandleAssociation hassoc = handleMap.get(request);
-        if (null == hassoc || null == hassoc.databaseName || !hassoc.databaseName.equals(catalog)) {
-            CS = request.prepare(connection);
-            hassoc = new HandleAssociation(catalog, CS);
-            HandleAssociation previous = handleMap.put(request, hassoc);
-            if (null != previous) {
-                previous.close();
+        try {
+            if (null == hassoc) {
+                CS = request.prepare(connection);
```

Comment:
This CallableStatement used to be closed when this method called previous.close() from line 273. However, we removed this part, and this CS will never be closed. The method the calls this method (getResultSetFromStoredProc) also doesn't clean up this returned CS. Is this CS intentionally left unclosed?